compose/ui/ui-test/src/desktopMain/kotlin/androidx/compose/ui/test/ComposeUiTest.desktop.kt

```diff
@@ -38,5 +36,40 @@ fun runDesktopComposeUiTest(
     effectContext: CoroutineContext = EmptyCoroutineContext,
     block: DesktopComposeUiTest.() -> Unit
 ) {
-    DesktopComposeUiTest(width, height, effectContext).runTest(block)
+    DesktopComposeUiTest(width, height, effectContext).runDesktopTest(block)
+}
+
+@ExperimentalTestApi
+class DesktopComposeUiTest(
+    width: Int = 1024,
+    height: Int = 768,
+    effectContext: CoroutineContext = EmptyCoroutineContext,
+    density: Density = Density(1f),
+) : SkikoComposeUiTest(width, height, effectContext, density) {
+
+    private val idlingResources = mutableSetOf<IdlingResource>()
+
+    override fun areAllResourcesIdle(): Boolean {
+        return synchronized(idlingResources) {
+            idlingResources.all { it.isIdleNow }
+        }
+    }
+
+    fun registerIdlingResource(idlingResource: IdlingResource) {
+        synchronized(idlingResources) {
+            idlingResources.add(idlingResource)
+        }
+    }
+
+    fun unregisterIdlingResource(idlingResource: IdlingResource) {
+        synchronized(idlingResources) {
+            idlingResources.remove(idlingResource)
+        }
+    }
+
+    fun runDesktopTest(block: DesktopComposeUiTest.() -> Unit) {
+        runTest {
+            block()
+        }
+    }
```

Comment:
This function looks a bit weird here. It's member of `DesktopComposeUiTest`, but you don't ever want to call it inside the test itself. The same actually applies to `SkikoComposeUiTest.runTest` too.

We could make it internal, or get rid of it completely, replacing it with
```
    with(DesktopComposeUiTest(width, height, effectContext)) {
        runTest { block() }
    }
```
`runTest` could just be made internal (because `SkikoComposeUiTest` is experimental).

Reply:
- SkikoComposeUiTest.runTest can't be internal because it's used in DesktopComposeTestRule (ui-test-junit4 module). 

- got rid of  `runDesktopTest `